sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -30,54 +30,61 @@ import org.apache.spark.sql.types._
  */
 object NestedColumnAliasing {
```

Comment:
I think a good first step is to improve the classdoc to explain in detail what we are doing here, with examples.

A good example is `DecorrelateInnerQuery`.

Reply:
I wasn't sure how much to add to the classdoc, but I added some more examples.